### PR TITLE
updated truffle to v5 and solidity to 0.5.3

### PR DESCRIPTION
**Pull request by starspire**
This PR contains upgrading truffle framework to v5 version and solidity language itself up to 0.5.3. Although no business changes were introduced to Gardener itself, many changes were needed to in order to make it worked with new major versions of language and framework. There is still not resolved problem with solidity coverage, which seems to not support solidity >= 0.5.0. Some more info on that problem [here](https://github.com/sc-forks/solidity-coverage/issues/309)

### CODE REVIEW

**Comment by starspire** [comment]
Fixes #9

**Comment by starspire** [comment]
I found the workaround how to force solidity-coverage to work with solidity >= 0.5. The drawback is that I was pushed to downgrade solidity from 0.5.3 to 0.5.0.

File: package.json [hunk 1]
```diff
@@ -23,20 +23,21 @@
...
+    "lodash": "4.17.11",
+    "openzeppelin-solidity": "2.3.0",
+    "solidity-coverage": "0.6.0-beta.5",
+    "ethlint": "1.2.4",
+    "truffle": "5.0.20",
+    "truffle-flattener": "1.3.0"
   },
-  "dependencies": {}
+  "dependencies": {
+    "dotenv": "8.0.0",
```
Comment (tcichowicz) [inline package.json]: Why are these in `dependencies`?

File: contracts/Oracle.sol [hunk 2]
```diff
@@ -21,41 +21,42 @@ contract Oracle is Authorizable {
...
+    event DataRequested(bytes32 indexed id, string url);
+    event DelayedDataRequested(bytes32 indexed id, string url, uint validFrom);
+    event RequestFulfilled(bytes32 indexed id, string value, uint errorCode);
 
     constructor(address _trustedServer) public {
         trustedServer = _trustedServer;
     }
 
-    function request(string _url) public onlyRole(AUTHORIZED_ROLE) returns(bytes32 id) {
+    function request(string memory _url) public onlyAuthorized() returns(bytes32 id) {
```
Comment (tcichowicz) [inline contracts/Oracle.sol]: Do you need parentheses when applying a parameterless modifier?

File: contracts/Migrations.sol [hunk 3]
```diff
@@ -1,4 +1,4 @@
-pragma solidity ^0.4.24;
+pragma solidity ^0.5.0;
```
Comment (tcichowicz) [inline contracts/Migrations.sol]: Why not apply 0.5.9?

Comment (starspire) [reply contracts/Migrations.sol]: It's a workaround to get solidity-coverage working. Although we explicitly set solidity compiler version to 0.5.9 it changes it during generated coverage to 0.5.3.

File: package.json (hunk 1 above)
Comment (starspire) [reply package.json]: We use them in truffle config which is detected by IDE as a non-dev file (which is particularly true  😄 ).

**Pull request merged**